NachoSpigot-Server/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java

```diff
@@ -541,6 +542,17 @@ public void setPullWhileLeashed(boolean pullWhileLeashed) {
     }
     // TacoSpigot end
 
+    // Nacho start
+    public void heal(float amount){
+        if(getHealth() + amount <= getMaxHealth()) getHandle().heal(amount);
+    }
+
+    public void heal(float amount, EntityRegainHealthEvent.RegainReason reason){
+        if(getHealth() + amount <= getMaxHealth())
+            getHandle().heal(amount, reason);
+    }
```

Comment:
```suggestion
    public void heal(float amount, EntityRegainHealthEvent.RegainReason reason){
        double max = getMaxHealth()-getHealth();
        if ((health < 0) || (health > max)) {
            throw new IllegalArgumentException("Heal amount must be between 0 and " + max + ", but was " + health
                + ". (attribute base value: " + this.getHandle().getAttributeInstance(GenericAttributes.maxHealth).b()
            + (this instanceof CraftPlayer ? ", player: " + this.getName() + ')' : ')'));
        }
        getHandle().heal(amount, reason);
    }
```

Reply:
Uhh, that part of my PRQ is already merged. I'm not sure how to separate these changes into a new one, though.